gr-blocks/lib/tagged_file_sink_impl.cc

```diff
@@ -63,12 +63,8 @@ tagged_file_sink_impl::tagged_file_sink_impl(size_t itemsize, double samp_rate)
                  io_signature::make(1, 1, itemsize),
                  io_signature::make(0, 0, 0)),
       d_itemsize(itemsize),
-      d_n(0),
```

Comment:
I don't understand why these need to be moved from the constructor into the class itself ... I thought it "normal" to set all class variables in the constructor ... what am I missing?

Reply:
Also: Why are some variables left in the constructor instantiation (d_itemsize, d_sample_rate) while others are not? Is it that those left behind are set from constructor arguments while the others are set in the class declaration? Hmmm ... not sure it really matters ... does it?

Reply:
Hmmm ... so my preference is to keep all settings in the constructor instantiation ... either above or here. Not setting in the class declaration itself. I don't see why Coverity doesn't like these ...

Reply:
Doesn't need to be moved. If not consting (which we aren't, here) then it's the same.

But since we have two constructors, where one forgot to set it, it seems better to me to set it in one place, than remember it in every constructor.

Coverity didn't like that in one constructor it was not set *at all*.

Reply:
Wait, this was actually a reply applicable to another change.

So you're saying if there's only one constructor you prefer set by constructor?

Reply:
Ah ... I didn't realize there were multiple constructors. OK yeah so if there are multiple, then I can see setting it in the class itself so-as to avoid the annoying duplication issue. When I was learning C++ a -long- time ago, I was taught to not set variables directly int the class itself, but rather somehow in the constructors. What I've always done is create an "init" function that's called by each constructor sometime during construction. It then sets variables as appropriate given the state of construction. In this case, it seems like all of the variables have defaults that are not other needed during construction, so why not set them in the class itself? I'm OK with this new way, but let's be consistent with -any- such variables in this class, please!

Reply:
> So you're saying if there's only one constructor you prefer set by constructor?

Yes that'd be my preference for a single constructor.

Reply:
OK since we're using C++11, we can use either constructor initialization or Declaration initialization. Both have benefits and drawbacks. I think we need to discuss this in more detail in your GREP to decide on which to use when and where.